src/core/spotifyApi.ts

```diff
@@ -1,20 +1,17 @@
-export const spotifyApi = (tkn?: string): spotifyApiFunction => {
-  const token: Token = {
-    current: tkn ?? '',
-  };
+import {
+  PersistentApiProperties,
+  QueryableApi,
+  QueryFunction,
+} from './utils/types';
 
-  return <T>(fn: QueryFunction<T>): T => {
-    if (!token.current) throw new Error('Token has not yet been provided');
-    return fn(token);
+export function spotifyApi(tkn: string): QueryableApi {
+  const props: PersistentApiProperties = {
+    token: tkn ?? '',
```

Comment:
Don't need nullish coalescing if the token cannot be null.

Preferred an explicit error throw for JS users trying to skip the token to explain what they need to do. (And test for it)